packages/api-http/source/controllers/transactions.ts

```diff
@@ -7,13 +7,13 @@ import { Controller } from "./controller";
 
 @injectable()
 export class TransactionsController extends Controller {
-	@inject(ApiDatabaseIdentifiers.TransactionRepository)
-	private readonly transactionRepository!: ApiDatabaseContracts.ITransactionRepository;
+	@inject(ApiDatabaseIdentifiers.TransactionRepositoryFactory)
+	private readonly transactionRepositoryFactory!: ApiDatabaseContracts.ITransactionRepositoryFactory;
 
 	public async index(request: Hapi.Request, h: Hapi.ResponseToolkit) {
 		const pagination = this.getQueryPagination(request.query);
 
-		const [transactions, totalCount] = await this.transactionRepository
+		const [transactions, totalCount] = await this.transactionRepositoryFactory()
```

Comment:
Can we initialize repositories on controller creation?

Reply:
As discussed, `QueryHelper` makes the repositories stateful. So a copy is needed per request (which should be cheap). Made a task to eventually improve it if needed.